src/scriptnum/scriptnum.cairo

```diff
@@ -72,15 +72,18 @@ pub mod ScriptNum {
         if result > MAX_INT32.into() || result < MIN_INT32.into() {
```

Comment:
I think changing `MIN_INT32` to -2147483647 would be sufficient, since this range check here would cover the issue. I think all uses of `try_into_num` should fail, even when `unwrap` is used. I think that one of the tests in `test_scriptnum_wrap_unwrap` will need to be changed to reflect this.